Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+
+        //This should correlate to the Orders list in Oanda.
+        private readonly IOrderProvider _orderProvider;
+
+        private readonly ConcurrentDictionary<long, DataType.Order> _cachedOpenOrdersByOandaOrderId;
+
+        /// <summary>
+        /// The QC User id, used for refreshing the session
+        /// </summary>
+        public int UserId { get; private set; }
+
+        /// <summary>
+        /// Access Token Access:
+        /// </summary>
+        public static string AccessToken { get; private set; }
+
+        /// <summary>
+        /// Refresh Token Access:
+        /// </summary>
+        public string RefreshToken { get; private set; }
+
+
+        /// <summary>
+        /// Creates a new Brokerage instance with the specified name
+        /// </summary>
+        /// <param name="name">The name of the brokerage</param>
+        public OandaBrokerage(string name, int userId) : base(name)
+        {
+            UserId = userId;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OandaBrokerage"/> class.
+        /// </summary>
+        /// <param name="orderProvider">The order provider.</param>
+        /// <param name="holdingsProvider">The holdings provider.</param>
+        /// <param name="accountId">The account identifier.</param>
+        public OandaBrokerage(IOrderProvider orderProvider, IHoldingsProvider holdingsProvider, int accountId)
+            : base("Oanda Brokerage")
+        {
+            _orderProvider = orderProvider;
+            _holdingsProvider = holdingsProvider;
+            AccountId = accountId;
+            _cachedOpenOrdersByOandaOrderId = new ConcurrentDictionary<long, DataType.Order>();
+            InstrumentSecurityTypeMap =  new Dictionary<string, SecurityType>(); 
+        }
+
+        /// <summary>
+        /// Initializes the instrument security type map.
+        /// </summary>
+        public void InitializeInstrumentSecurityTypeMap()
+        {
+            InstrumentSecurityTypeMap = MapInstrumentToSecurityType(GetInstrumentsAsync());
+        }
+
+        /// <summary>
+        /// Returns true if we're currently connected to the broker
+        /// </summary>
+        public override bool IsConnected
+        {
+            get { return _issuedAt + _lifeSpan > DateTime.Now; }
+        }
+
+        /// <summary>
+        /// Gets all open orders on the account. 
+        /// NOTE: The order objects returned do not have QC order IDs.
+        /// </summary>
+        /// <returns>The open orders returned from Oanda</returns>
+        public override List<Order> GetOpenOrders()
+        {
+            var oandaOrders = GetOrderList();
+
+            var orderList = oandaOrders.Select(ConvertOrder).ToList();
+            foreach (var openOrder in orderList)
+            {
+                _cachedOpenOrdersByOandaOrderId[openOrder.Id] = oandaOrders.FirstOrDefault(oo => oo.id == openOrder.Id);
```

Comment:
Yes it's misleading :). Nonetheless because cachedOpenOrdersByOandaOrderId isn't getting used, I've removed this line.